rust/core-lib/src/editor.rs

```diff
@@ -691,171 +425,156 @@ impl Editor {
             }
 
             match last_selection_region(view.sel_regions()) {
-                Some(_) => self.insert(view, chars),
+                Some(_) => self.add_delta(edit_ops::insert(&self.text, view.sel_regions(), chars)),
                 None => return,
             };
         }
     }
 
-    fn transform_text<F: Fn(&str) -> String>(&mut self, view: &View, transform_function: F) {
-        let mut builder = DeltaBuilder::new(self.text.len());
+    pub(crate) fn do_edit(
+        &mut self,
+        view: &mut View,
+        kill_ring: &mut Rope,
+        config: &BufferItems,
+        cmd: BufferEvent,
+    ) {
+        use self::BufferEvent::*;
 
-        for region in view.sel_regions() {
-            let selected_text = self.text.slice_to_cow(region);
-            let interval = Interval::new(region.min(), region.max());
-            builder.replace(interval, Rope::from(transform_function(&selected_text)));
-        }
-        if !builder.is_empty() {
-            self.this_edit_type = EditType::Other;
-            self.add_delta(builder.build());
-        }
-    }
+        let regions = view.sel_regions();
 
-    /// Changes the number(s) under the cursor(s) with the `transform_function`.
-    /// If there is a number next to or on the beginning of the region, then
-    /// this number will be replaced with the result of `transform_function` and
-    /// the cursor will be placed at the end of the number.
-    /// Some Examples with a increment `transform_function`:
-    ///
-    /// "|1234" -> "1235|"
-    /// "12|34" -> "1235|"
-    /// "-|12" -> "-11|"
-    /// "another number is 123|]" -> "another number is 124"
-    ///
-    /// This function also works fine with multiple regions.
-    fn change_number<F: Fn(i128) -> Option<i128>>(&mut self, view: &View, transform_function: F) {
-        let mut builder = DeltaBuilder::new(self.text.len());
-        for region in view.sel_regions() {
-            let mut cursor = WordCursor::new(&self.text, region.end);
-            let (mut start, end) = cursor.select_word();
-
-            // if the word begins with '-', then it is a negative number
-            if start > 0 && self.text.byte_at(start - 1) == (b'-') {
-                start -= 1;
+        match cmd {
+            Delete { movement, kill } => {
```

Comment:
reading through this patch, I think my personal inclination would be to keep using the separate `do_$thing` methods, and then just in those methods call out to `edit_ops`; a major advantage of this approach is that it makes this diff easier to read and verify.

Reply:
I've reversed this.
I didn't want multiple related functions with the same name doing different things when I made this. For now, the functions in `editor.rs` that call `edit_ops.rs` functions have the `do_`-prefix, while the functions in `edit_ops.rs` do not have that prefix.